tests/test_api_build.py

```diff
@@ -1178,20 +1177,14 @@ def test_failed_recipe_leaves_folders(testing_config, testing_workdir):
         api.build(m)
     assert os.path.isdir(m.config.build_folder), 'build folder was removed'
     assert os.listdir(m.config.build_folder), 'build folder has no files'
+
     # make sure that it does not leave lock files, though, as these cause permission errors on
     #    centralized installations
-    any_locks = False
-    locks_list = set()
-    locale.getpreferredencoding(False)
-    for lock in locks:
-        if os.path.isfile(lock.lock_file):
-            any_locks = True
-            dest_path = base64.b64decode(os.path.basename(lock.lock_file))
-            if hasattr(dest_path, 'decode'):
-                dest_path = dest_path.decode(sys.getfilesystemencoding(), errors='backslashreplace')
-            locks_list.add((lock.lock_file, dest_path))
-    assert not any_locks, "remaining locks:\n{}".format('\n'.join('->'.join((l, r))
-                                                                for (l, r) in locks_list))
+    locks = {lock.lock_file for lock in locks if os.path.isfile(lock.lock_file)}
+    assert not locks, "\n".join(
+        "remaining locks:",
+        *(f"  {lock}" for lock in locks.items()),
+    )
```

Comment:
hm good point, doing `not locks` will also be sufficient:

<img width="590" alt="Screen Shot 2022-12-22 at 11 51 13 AM" src="https://user-images.githubusercontent.com/4546435/209195914-687d2ad2-5d1d-42b2-9448-7cb214eb8767.png">

```suggestion
    assert not {lock.lock_file for lock in locks if os.path.isfile(lock.lock_file)}
```

Reply:
(This might not be terribly important in this case, but for longer lists/entries you get a proper diff if you use `==`. With `not ...` those longer entries will actually just be truncated instead.)

Reply:
We run `pytest -v` in CI so doesn't that already show us the full diff?

Reply:
`assert not ...` will show you `AssertionError: assert not ...<truncated>`.
`assert [] == ...` will show you `AssertionError: assert [] == ...<truncated>` and additionally a diff (also truncated).
(A non-truncated "full diff" gets output if you run `pytest -vv`.)